wrangler-storage/src/main/java/io/cdap/wrangler/dataset/schema/SchemaFile.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.wrangler.dataset.schema;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import org.apache.commons.lang.builder.EqualsBuilder;
+
+import java.util.Map;
+
+/**
+ * The deserialized representation of a JSONSchema file.
+ */
+public class SchemaFile {
+  private String description;
+  private Discriminator discriminator;
+  private MetaData metadata;
+  private Map<String, JsonElement> definitions;
+
+  public SchemaFile(){}
+  public String getDescription() {
+    return description;
+  }
+  public Discriminator getDiscriminator() {
+    return discriminator;
+  }
+  public MetaData getMetadata() {
+    return metadata;
+  }
+  public Map<String, JsonElement> getDefinitions() {
+    return definitions;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    SchemaFile def = (SchemaFile) o;
+    return new EqualsBuilder()
+        .append(description, def.description)
+        .append(discriminator, def.discriminator)
+        .append(metadata, def.metadata)
+        .append(definitions, def.definitions)
+        .isEquals();
+  }
+
+  /**
+   * Accessor for instantiating a builder for constructing a {@SchemaFile}.
+   * @return a builder to populate.
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  /**
+   * Builds a {@SchemaFile}
+   */
+  public static class Builder {
+    private String jsonSchemaDef;
+    private static final Gson GSON = new Gson();
+
+    Builder(){}
+    public Builder setJson(String json) {
```

Comment:
builders are used to set the individual fields. If gson is being used to deserialize an object, it should be used directly by the caller and not through a builder class here. We don't want gson classes here in the proto objects.